libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java

```diff
@@ -80,7 +80,7 @@ public class XContentParserTests extends OpenSearchTestCase {
         () -> randomAlphaOfLengthBetween(1, SmileXContent.DEFAULT_MAX_STRING_LEN),
         /* YAML parser limitation */
         XContentType.YAML,
-        () -> randomRealisticUnicodeOfCodepointLengthBetween(1, 3140000)
```

Comment:
Sadly the `randomRealisticUnicodeOfCodepointLengthBetween` didn't play well with YAML parser limits

Reply:
Does this method actually generate a string with more than 3145728 code points as suggested by the error above? Or is the 3145728 limit at the document level and we end up adding multiple large fields resulting in a document with more than 3145728 code points?

Reply:
> Or is the 3145728 limit at the document level and we end up adding multiple large fields resulting in a document with more than 3145728 code points?

I looked at the test again and it seems we only add one field to the document, so this doesn't make sense. I'm still not clear why the unicode fails (and therefore not clear why switching to ASCII fixes it).

Reply:
I have not looked deeply to be fair, but it seems like there are differences in the way `randomRealisticUnicodeOfCodepointLengthBetween` and YAMLParser calculate code points

Reply:
It looks like `randomRealisticUnicodeOfCodepointLengthBetween` can generate characters beyond the BMP range, which require 2 UTF-16 characters to represent one code point. That is correct, but the YAML limit seems to be implemented against the `String.length()` value, which can be double the number of code points when non-BMP characters are involved.

The change to ASCII will indeed fix this. We could also supply `3140000 / 2` as the limit and keep the unicode. I suppose either way is fine.

Reply:
Thanks @andrross 

> That is correct, but the YAML limit seems to be implemented against the String.length() value,

This is very surprising, the exception parser raises states specifically "maximum codepoints limit" (not characters or alike), weird. But thanks for looking it up!